met-web/src/components/common/Modals/Schedule/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useContext, useState } from 'react';
+import { Grid, Stack, useMediaQuery, Theme, TextField, Modal } from '@mui/material';
+import { modalStyle, PrimaryButton, SecondaryButton, MetHeader1, MetBody, MetLabel } from 'components/common';
+import dayjs, { Dayjs } from 'dayjs';
+import { useAppDispatch, useAppSelector } from 'hooks';
+import { EngagementStatus } from 'constants/engagementStatus';
+import { DesktopDatePicker } from '@mui/x-date-pickers/DesktopDatePicker';
+import { TimePicker } from '@mui/x-date-pickers/TimePicker';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { ActionContext } from 'components/engagement/view/ActionContext';
+import { openNotification } from 'services/notificationService/notificationSlice';
+
+interface ScheduleModalProps {
+    reschedule: boolean;
+    open: boolean;
+    updateModal: (open: boolean) => void;
+}
+
+const ScheduleModal = ({ reschedule, open, updateModal }: ScheduleModalProps) => {
+    const isSmallScreen = useMediaQuery((theme: Theme) => theme.breakpoints.down('sm'));
+    const [scheduledDate, setScheduledDate] = useState<Dayjs | null>(dayjs('2014-08-18T21:11:54'));
+    const { savedEngagement, publishEngagement } = useContext(ActionContext);
+    const dispatch = useAppDispatch();
+
+    const handleChange = (newDate: Dayjs | null) => {
+        if (newDate != null) {
+            console.log('DATE DATA::::' + newDate.format('DD-MM-YYYY HH:mm:ss'));
+            setScheduledDate(newDate);
+        }
+    };
+
+    const handleSchedule = async () => {
+        if (savedEngagement.surveys.length === 0) {
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Please add a survey to the engagement before publishing it',
+                }),
+            );
+            return;
+        }
+        await publishEngagement({
```

Comment:
method name should be scheduleEngagement

Reply:
all changes have been resolved :)